internal/hcs/system.go

```diff
@@ -328,27 +321,29 @@ func (computeSystem *System) Terminate() (err error) {
 	}()
 
 	if computeSystem.handle == 0 {
-		return makeSystemError(computeSystem, "Terminate", "", ErrAlreadyClosed, nil)
+		return false, nil
 	}
 
 	var resultp *uint16
 	syscallWatcher(computeSystem.logctx, func() {
 		err = hcsTerminateComputeSystem(computeSystem.handle, "", &resultp)
 	})
 	events := processHcsResult(resultp)
-	if err != nil && err != ErrVmcomputeAlreadyStopped {
-		return makeSystemError(computeSystem, "Terminate", "", err, events)
+	if IsAlreadyStopped(err) || IsNotExist(err) {
```

Comment:
Interesting. Moving these in, since it is before wrap it seems like you can instead just write:
```go
    switch err {
    case ErrVmcomputeAlreadyStopped, ErrComputeSystemDoesNotExist:
        return false, nil
    case nil, ErrVmcomputeOperationPending:
        return true, nil
    default:
        return false, makeSystemError(computeSystem, "Terminate", "", err, events)
    }
```

Reply:
Not sure if its better but it seems more clear since we can match the exact `syscall.Errno` from the native call.